### PR TITLE
Bug fix: Old API deployment fails due to not having the throttlingLimit Property

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [ ] Assigned the project
- [x] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/model/api_yaml_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (slahirucd7) [inline adapter/internal/oasparser/model/api_yaml_test.go:2]: `Copyright (c) 2023, WSO2 LLC. `

**Review by slahirucd7 — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230328.7

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : dev-deployment-v2 : 20230328.7

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230328.7

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230328.8

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230328.8

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230328.8

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230328.2

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20230328.2

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230328.2

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20230328.2

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230328.2

**Comment by choreo-cicd** [comment]
[failed] Dataplane(CentralIN) cluster : prod-deployment-v2 : 20230328.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : prod-deployment-v2 : 20230328.3

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20230328.4

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CentralIN) cluster : prod-deployment-v2 : 20230328.4

**Pull request merged**